presto-druid/src/test/java/io/prestosql/plugin/druid/TestingDruidServer.java

```diff
@@ -0,0 +1,286 @@
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("common.runtime.properties").getPath()),
+                            "/opt/druid/conf/druid/cluster/_common/common.runtime.properties")
+                    .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("historical.config").getPath()),
+                            "/opt/druid/conf/druid/cluster/data/historical/runtime.properties")
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("historical-jvm.config").getPath()),
+                            "/opt/druid/conf/druid/cluster/data/historical/jvm.config")
+                    .waitingFor(Wait.forHttp("/status/selfDiscovered"));
+            historical.start();
+
+            this.middleManager = new GenericContainer(DRUID_DOCKER_IMAGE)
+                    .withExposedPorts(DRUID_MIDDLE_MANAGER_PORT)
+                    .withNetwork(network)
+                    .withCommand("middleManager")
+                    .withWorkingDirectory("/opt/druid")
+                    .dependsOn(zookeeper, coordinator)
+                    .withClasspathResourceMapping("druid-test-storage/", "/opt/druid/var", BindMode.READ_WRITE)
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("common.runtime.properties").getPath()),
+                            "/opt/druid/conf/druid/cluster/_common/common.runtime.properties")
+                    .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("middleManager.config").getPath()),
+                            "/opt/druid/conf/druid/cluster/data/middleManager/runtime.properties")
+                    .withCopyFileToContainer(
+                            forHostPath(getResource("middleManager-jvm.config").getPath()),
+                            "/opt/druid/conf/druid/cluster/data/middleManager/jvm.config")
+                    .waitingFor(Wait.forHttp("/status/selfDiscovered"));
+            middleManager.start();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public String getHostWorkingDirectory()
+    {
+        return hostWorkingDirectory;
+    }
+
+    @Override
+    public void close()
+    {
+        try (Closer closer = Closer.create()) {
+            closer.register(broker::stop);
+            closer.register(historical::stop);
+            closer.register(middleManager::stop);
+            closer.register(coordinator::stop);
+            closer.register(zookeeper::stop);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        finally {
+            cleanDirectory(new File(hostWorkingDirectory));
```

Comment:
Register this in closer as well:

```
 closer.register(() -> deleteRecursively(hostWorkingDirectory, ALLOW_INSECURE));
```

(first, so it's get called last)

Reply:
Thanks for the tip on MoreFiles. I think ```MoreFiles.deleteDirectoryContents``` is a better option.